domains/runtime/evm/src/lib.rs

```diff
@@ -667,21 +667,20 @@ fn extract_xdm_proof_state_roots(
 fn extract_signer_inner<Lookup>(
     ext: &UncheckedExtrinsic,
     lookup: &Lookup,
-) -> Option<opaque::AccountId>
+) -> Option<Result<AccountId, TransactionValidityError>>
```

Comment:
its weird seeing TransactionValidityError while this function essentially just extracts signer

Reply:
@vedhavyas This is because `check_self_contained` returns `TransactionValidityError` and `Lookup::lookup` returns `LookupError` which has `impl From<LookupError> for TransactionValidityError` so it made sense to return the `TransactionValidityError` here. 

Do not want to wrap it as this is private function and caller would be wrapping or ignoring the error.

Reply:
We would need anyway need to propagate this error to client through runtime_api as well, so maybe we can create a specific error type that impls Lookup as well, no?

Reply:
If we need to propagate it outside then it make sense to have dedicated error type. But, I would like to address this in another PR. This is fundamentally in scope of the new issue I am talking about opening here: https://github.com/subspace/subspace/pull/2071#discussion_r1356575995